### PR TITLE
V2 Funding history tab

### PR DESCRIPTION
**Pull request by johnnyd-eth**
## What does this PR do and why?

_Provide a description of what this PR does. Link to any relevant GitHub issues,
Notion tasks or Discord discussions._

## Screenshots or screen recordings

_If applicable, provide screenshots or screen recordings to demonstrate the
changes._

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).

### CODE REVIEW

**Review by aeolianeth — CHANGES_REQUESTED** [review]
@johnnyd-eth hell yeah nice job figuring this out!

Mostly appears to work for me, but I encountered a couple issues and have some suggestions:

### potential bugs

<details><summary>runtime exception</summary>

I got an error looking at `/v2/p/2`

Screen Shot 2022-04-29 at 10 51 56 pm

</details>

<details><summary>Not all FC's loading</summary>

Looking at http://localhost:3000/#/v2/p/12, funding cycles 3 2 and 1 never seem to load.

Screen Shot 2022-04-29 at 10 54 34 pm

</details>

### Suggestions

- give the section a `max-height` and make it scrollable. http://localhost:3000/#/v2/p/12 has many funding cycles that blow up the content of the page
- Limit list to 5 funding cycles. Load 5 at a time for better UX. Maybe this could be a follow-up issue.

File: src/components/v2/V2Project/V2FundingCycleSection/FundingCycleHistory.tsx [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+      while (!previousReconfiguration.eq(BigNumber.from(0))) {
+        const previousReconfigurationFirstFundingCycle: V2FundingCycle =
+          (await contracts?.JBFundingCycleStore.get(
+            projectId,
+            previousReconfiguration,
+          )) as V2FundingCycle
+
+        if (previousReconfigurationFirstFundingCycle) {
+          // Add it to the start of list
+          ;[previousReconfigurationFirstFundingCycle].concat(
```
[the remark this answers is not in the document]
Comment (johnnyd-eth) [reply src/components/v2/V2Project/V2FundingCycleSection/FundingCycleHistory.tsx:154]: No 😅

File: src/components/v2/V2Project/V2FundingCycleSection/FundingCycleHistory.tsx [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+            <>
+              <Trans>
+                {formatWad(usedDistributionLimit, { precision: 2 })} withdrawn
+              </Trans>
+            </>
+          )}
+        </div>
+      </Space>
+
+      <div style={{ flex: 1 }}></div>
```
[the remark this answers is not in the document]
Comment (johnnyd-eth) [reply src/components/v2/V2Project/V2FundingCycleSection/FundingCycleHistory.tsx:105]: Nothing!

File: src/components/v2/V2Project/V2FundingCycleSection/FundingCycleHistory.tsx [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+    useContext(V2ProjectContext)
+  const { contracts } = useContext(V2UserContext)
+
+  const [selectedIndex, setSelectedIndex] = useState<number>()
+  const [pastFundingCycles, setPastFundingCycles] = useState<V2FundingCycle[]>(
+    [],
+  )
+
+  useEffect(() => {
+    const loadPastFundingCycles = async () => {
```
[the remark this answers is not in the document]
Comment (johnnyd-eth) [reply src/components/v2/V2Project/V2FundingCycleSection/FundingCycleHistory.tsx:128]: > **suggestion** this function is quite large and hard to follow. WDYT about isolating the logic into smaller functions to make it more readable/testable?

TBH I don't think it would help very much.

**Comment by johnnyd-eth** [comment]
All comments and bugs you've mentioned here are fixed. Confident in this now

**Review by aeolianeth — APPROVED** [review]
